managed-identities.yaml

```diff
@@ -0,0 +1,17 @@
+version: v1.0.0
+secrets:
+  - id: name
+    keyvault: https://myacbvault2.vault.azure.net/secrets/SampleSecret
+steps:
+  - cmd: bash -c 'if [ -z "$MY_SECRET" ]; then echo "Secret not resolved"; else echo "Secret resolved"; fi'
+    env: 
+      - MY_SECRET='{{.Secrets.name}}' 
+
+  # Build/Push the website to source registry
+  - cmd: docker build -t {{.Run.Registry}}/my-website:{{.Run.ID}} https://github.com/Azure-Samples/aci-helloworld.git
+  - push: 
+    - "{{.Run.Registry}}/my-website:{{.Run.ID}}"
+  
+  # Login to Azure and list the tags to verify if we have the Image!
+  - cmd: microsoft/azure-cli az login --identity
```

Comment:
@shahzzzam - Sam, this KV/login example uses a user-defined identity as explained in your doc. Is it OK for the task step to login to az CLI with a system-managed identity (presumably derived from the task?), or should it be logging in with the user-assigned identity?

Reply:
@dlepow it doesn't matter.. you can either assign a user assigned identity or a system assigned identity to the task. As long as you the assign identity a role in Azure, we will be able to login. Overall, in any example, you can use user-assigned or system-assigned identity. Hope this helps.